File: msme_intel-main/msme-demo/lib/types/mosi.ts
// msme-demo/lib/types/mosi.ts

export type CEEDTag = 'Core' | 'Efficiency' | 'Expansion' | 'Disrupt';

export interface TranscriptParagraph {
    id: string;
    text: string;
    speaker: string;
    timestamp: number;
    status: 'Approved' | 'Hidden' | 'Pending';
    comment?: string;
}

export interface MosiStakeholder {
    id: string;
    company_id?: string;
    user_id?: string;
    name: string;
    role?: string;
    email?: string;
    phone?: string;
    linkedin?: string;
    company_name?: string;
    sector?: string;
    employees?: string;
    revenue?: string;
    geography?: string;
    domain?: string;
    address?: string;
    pincode?: string;
    created_at?: string;
}

export interface MosiSession {
    id: string;
    stakeholder_id: string;
    user_id?: string;
    company_id?: string;
    companyId?: string; // Support for legacy camelCase in demo data
    status: 'Scheduled' | 'Recording' | 'Review' | 'Published' | string;
    date: string;
    duration: number; // in seconds
    audio_settings?: {
        audio: boolean;
        video: boolean;
    };
    summary?: string;
    problem_summary?: string;
    potential_roi?: string;
    tech_stack_recommended?: string[];
    transcript: (TranscriptParagraph | { speaker: string; time: string; text: string; opportunity?: boolean })[];
    recording_url?: string;
    created_at?: string;
    metadata?: {
        stakeholder: string;
        company: string;
        session_name: string;
    };
}

export interface MosiOpportunity {
    id: string;
    session_id: string;
    timestamp: number;
    title: string;
    description?: string;
    tag: CEEDTag;
    
    // Rubric Scoring (1-4)
    problem_clarity: number;
    budget_score: number;
    
    // Logistics & Matching
    is_paid: boolean;
    reward_amount?: string;
    working_hours?: number;
    duration_commitment?: string;
    
    origin: 'Customer' | 'Interviewer';
    actively_seeking: boolean;
    
    // Talent Attributes
    skillset: string[];
    toolset: string[];
    mindset: string[];
    
    // Assessment Matrix
    assessment_matrix: {
        clarity: number;
        awareness: number;
        attempts: number;
        intensity: number;
    };
    
    notes?: string;
    status: 'Approved' | 'Hidden' | 'Pending';
    comment?: string;
    created_at?: string;
}

export interface MosiEvidence {
    id: string;
    session_id: string;
    opportunity_id?: string;
    type: 'image' | 'video' | 'link' | 'file';
    url: string;
    title?: string;
    timestamp?: number;
    created_at?: string;
}

export interface UnifiedOpportunity {
    id: string;
    session_id?: string;
    company_id?: string;
    title: string;
    description?: string;
    ceed_tag: string;
    problem_clarity: number;
    budget_availability: number;
    is_paid_project: boolean;
    budget_amount: number;
    working_hours_defined: number;
    owner_origin: string;
    customer_actively_seeking: boolean;
    skillset: string[];
    toolset: string[];
    mindset: string[];
    duration_weeks: number;
    sync_status: string;
}

File: msme_intel-main/backend/app/models/mosi.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from uuid import UUID
from datetime import datetime

class MosiStakeholder(BaseModel):
    id: Optional[UUID] = None
    company_id: Optional[UUID] = None
    user_id: Optional[UUID] = None
    name: str
    role: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    linkedin: Optional[str] = None
    company_name: Optional[str] = None
    sector: Optional[str] = None
    employees: Optional[str] = None
    revenue: Optional[str] = None
    geography: Optional[str] = None
    domain: Optional[str] = None
    address: Optional[str] = None
    pincode: Optional[str] = None
    created_at: Optional[datetime] = None

class TranscriptParagraph(BaseModel):
    id: str
    text: str
    speaker: str
    timestamp: float
    status: str = "Pending"
    comment: Optional[str] = None

class MosiSession(BaseModel):
    id: Optional[UUID] = None
    stakeholder_id: UUID
    user_id: Optional[UUID] = None
    company_id: Optional[UUID] = None
    status: str = "Review"
    date: Optional[str] = None
    duration: int = 0
    audio_settings: Dict[str, bool] = {"audio": True, "video": True}
    summary: Optional[str] = None
    transcript: List[TranscriptParagraph] = []
    recording_url: Optional[str] = None
    created_at: Optional[datetime] = None

class AssessmentMatrix(BaseModel):
    clarity: int = 2
    awareness: int = 2
    attempts: int = 2
    intensity: int = 2

class MosiOpportunity(BaseModel):
    id: Optional[UUID] = None
    session_id: UUID
    timestamp: int
    title: str
    description: Optional[str] = None
    tag: str # Core, Efficiency, Expansion, Disrupt
    problem_clarity: int = 2
    budget_score: int = 2
    is_paid: bool = False
    reward_amount: Optional[str] = None
    working_hours: Optional[int] = None
    duration_commitment: Optional[str] = None
    origin: str = "Customer"
    actively_seeking: bool = True
    skillset: List[str] = []
    toolset: List[str] = []
    mindset: List[str] = []
    assessment_matrix: AssessmentMatrix = AssessmentMatrix()
    notes: Optional[str] = None
    status: str = "Pending"
    comment: Optional[str] = None

class MosiEvidence(BaseModel):
    id: Optional[UUID] = None
    session_id: UUID
    opportunity_id: Optional[UUID] = None
    type: str
    url: str
    title: Optional[str] = None
    timestamp: Optional[int] = None

class UnifiedOpportunity(BaseModel):
    id: Optional[UUID] = None
    session_id: Optional[UUID] = None
    company_id: Optional[UUID] = None
    title: str
    description: Optional[str] = None
    ceed_tag: str
    problem_clarity: int = 2
    budget_availability: int = 2
    is_paid_project: bool = False
    budget_amount: float = 0.0
    working_hours_defined: int = 0
    owner_origin: str = "customer"
    customer_actively_seeking: bool = True
    skillset: List[str] = []
    toolset: List[str] = []
    mindset: List[str] = []
    duration_weeks: int = 4
    sync_status: str = "synced"
